gcd/src/main/java/org/spine3/server/storage/datastore/DsAggregateStorage.java

```diff
@@ -199,18 +145,19 @@ protected void writeRecord(I id, AggregateStorageRecord record) {
         final Entity incompleteEntity = Entities.messageToEntity(record, key);
         final Entity.Builder builder = Entity.newBuilder(incompleteEntity);
         DatastoreProperties.addAggregateIdProperty(stringId, builder);
+        builder.set(AGGREGATE_TYPE, stateTypeName.value());
         datastore.createOrUpdate(builder.build());
     }
 
     @Override
-    protected Iterator<AggregateStorageRecord> historyBackward(I id) {
+    protected Iterator<AggregateEventRecord> historyBackward(I id) {
         checkNotNull(id);
 
         final String idString = idToString(id);
         final Query<Entity> query = Query.newEntityQueryBuilder()
-                                    .setKind(KIND)
-                                    .setFilter(StructuredQuery.PropertyFilter.eq(aggregate_id.toString(), idString))
-                                    .build();
+                                         .setKind(KIND)
+                                         .setFilter(StructuredQuery.PropertyFilter.eq(aggregate_id.toString(), idString))
```

Comment:
Can we have a wrapping method for building the filter? We have a number of such code blocks here and there.